### PR TITLE
Add code, unit test and test data for ip to node type mapping

### PR DESCRIPTION
**Pull request by raghusarangapani**
This PR adds code, unit test and test data for ip to mode type mapping.
Refer https://github.com/couchbaselabs/mobile-testkit/issues/831 for the original issue.

### CODE REVIEW

File: libraries/utilities/generate_clusters_from_pool.py [hunk 1]
```diff
@@ -190,13 +338,17 @@ def get_ips(pool_file="resources/pool.json"):
     with open(pool_file) as f:
         pool_dict = json.loads(f.read())
         ips = pool_dict["ips"]
+        ip_to_node_type = []
+
+        if "ip_to_node_type" in pool_dict.keys():
```
Comment (sethrosetter) [inline libraries/utilities/generate_clusters_from_pool.py]: You can use also check using if `"ip_to_node_type" in pool_dict:` but whatever you think is best.

File: mobile_testkit_tests/test_generate_clusters_from_pool.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    IPS_POOL_FILE = TEST_DATA + "/mock_pool_ips.json"
+    CLUSTER_CONFIG_DIR = TEST_DATA + "/cluster_configs"
+
+    # Clear the cluster_configs folder if there are any files
+    # Else create a new folder
+    if os.path.exists(CLUSTER_CONFIG_DIR):
+        os.chdir(CLUSTER_CONFIG_DIR)
+        filesToRemove = [f for f in os.listdir(CLUSTER_CONFIG_DIR)]
+        for f in filesToRemove:
+            os.unlink(f)
```
Comment (raghusarangapani) [inline mobile_testkit_tests/test_generate_clusters_from_pool.py]: Done

File: mobile_testkit_tests/test_generate_clusters_from_pool.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+import json
+
+from libraries.utilities.generate_clusters_from_pool import generate_clusters_from_pool
+
+def test_generate_clusters_from_pool_ips():
+    # Function to test pool.json with only IPs
+    cwd = os.getcwd()
+    TEST_DATA = cwd + "/mobile_testkit_tests/test_data"
+    IPS_POOL_FILE = TEST_DATA + "/mock_pool_ips.json"
+    CLUSTER_CONFIG_DIR = TEST_DATA + "/cluster_configs"
```
Comment (raghusarangapani) [inline mobile_testkit_tests/test_generate_clusters_from_pool.py]: Done

File: mobile_testkit_tests/test_generate_clusters_from_pool.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    # Else create a new folder
+    if os.path.exists(CLUSTER_CONFIG_DIR):
+        os.chdir(CLUSTER_CONFIG_DIR)
+        filesToRemove = [f for f in os.listdir(CLUSTER_CONFIG_DIR)]
+        for f in filesToRemove:
+            os.unlink(f)
+
+        os.chdir(cwd)
+    else:
+        os.mkdir(CLUSTER_CONFIG_DIR)
```
Comment (raghusarangapani) [inline mobile_testkit_tests/test_generate_clusters_from_pool.py]: Done

File: libraries/utilities/generate_clusters_from_pool.py (hunk 1 above)
Comment (raghusarangapani) [inline libraries/utilities/generate_clusters_from_pool.py]: Done

File: mobile_testkit_tests/test_generate_clusters_from_pool.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    # Run tests with mock_pool_ips.json for backward compatibility
+    generate_clusters_from_pool(IPS_POOL_FILE)
+
+    # Verification
+    # mock_pool_ips.json will generate 22 files ansible+json
+    assert len([name for name in os.listdir(CLUSTER_CONFIG_DIR)])== 22
+
+    # We will check 2 files for content 1cbs/1cbs.json
+    config = configparser.ConfigParser()
+    config.read(CLUSTER_CONFIG_DIR + "/1cbs")
```
Comment (raghusarangapani) [inline mobile_testkit_tests/test_generate_clusters_from_pool.py]: Done

File: mobile_testkit_tests/test_generate_clusters_from_pool.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+    # Check only 1 couchbase server is configured in 1cbs
+    cbs = config["couchbase_servers"]
+    pool = config["pool"]
+    sgs = config["sync_gateways"]
+    sga = config["sg_accels"]
+    lg = config["load_generators"]
+    lb = config["load_balancers"]
```
Comment (raghusarangapani) [inline mobile_testkit_tests/test_generate_clusters_from_pool.py]: Done

File: mobile_testkit_tests/test_generate_clusters_from_pool.py [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+    # Check only 1 couchbase server is configured in 1cbs.json
+    with open(CLUSTER_CONFIG_DIR + "/1cbs.json") as data_file:
+        data = json.load(data_file)
+
+    assert len(data["couchbase_servers"]) == 1
+    assert len(data["hosts"]) == 6
+    assert len(data["sync_gateways"]) == 0
+    assert len(data["sg_accels"]) == 0
+    assert len(data["load_generators"]) == 0
+    assert len(data["load_balancers"]) == 0
```
Comment (raghusarangapani) [inline mobile_testkit_tests/test_generate_clusters_from_pool.py]: This code already exists in test_generate_clusters_from_pool_ip_to_node.

    # Verify that the IP from 1cbs is actually a defined as
    # a couchbase_servers in mock_pool_ip_to_node_type.json
    assert pool_data["ip_to_node_type"][cbs_ip] == "couchbase_servers"

    # Verify that the IP from 1cbs is actually a defined as
    # a couchbase_servers in mock_pool_ip_to_node_type.json
    assert pool_data["ip_to_node_type"][cbs_ip] == "couchbase_servers"

File: libraries/utilities/generate_clusters_from_pool.py [hunk 8]
```diff
@@ -40,10 +42,15 @@ def write_config(config, pool_file):
         )
         return
 
+    # Check for number of IPs versus number of IPs in ip_to_node_type
+    if ip_to_node_type and len(ip_to_node_type) != len(ips):
+        log_error("Number of IPs in resources/pool:ips and ip_to_node_type do not match. Exiting ...")
+        sys.exit(1)
```
Comment (tleyden) [inline libraries/utilities/generate_clusters_from_pool.py]: Instead of exiting, can you do `raise Exception("Number of IPs ...")`

Comment (raghusarangapani) [inline libraries/utilities/generate_clusters_from_pool.py]: Done

**Comment by sethrosetter** [comment]
Merging to unblock perf testing. @tleyden and I will be doing post merge reviews.

**Pull request merged**